src/spi.rs

```diff
@@ -482,70 +548,87 @@ impl<'d> SpiMasterDriver<'d> {
         let flush_result = bus.flush();
```

Comment:
i just realized we have a problem here using this transaction Fn with software managed Chip select. By calling finish the driver pulls the internal_cs high but we still write a complete block of 32 clk cycles out of the bus after finishing, this is not acceptable for devices that dont have an  cs / or have an managed cs.

Reply:
on the finish Fn not the flush Fn

Reply:
we writing garbage to the bus, the user is not aware of this

Reply:
why was the finish Fn introduced in the first place ? was there some problem getting cs lane high again?

Reply:
we have our hardware_cs flag -> one easy comparison would fix this by if hardware_cs.is_true() { call_to_finish} what you think ? you have to now that you already loose flat 10us because you call the call_to_finish function anyway.

Reply:
> was there some problem getting cs lane high again?

Yes. Unfortunately once the Spi peripheral has been give a pin, it prevents the pin from being changed. So using the `PinDriver` for example to toggle the CS pin won't work, as the peripheral keeps the pin low/high until it is told to change it.
So an empty transaction without the `KEEP_CS_ACTIVE` flag does the trick (on ESP32 at least).
I'm fairly certain I left a comment in the code explaining this.

Reply:
one comparison is much faster than that 10us :smile:

Reply:
and we could additional make it behind a flag for esp32 / s3 /c3 if this workaround is only needed for the esp32

Reply:
> one easy comparison would fix this by if hardware_cs.is_true() { call_to_finish}

Yes this would work. It's the point of the `hardware_cs` flag I guess.

> and we could additional make it behind a flag for esp32 if this workaround is only needed for the esp32 ( call to the finish Fn)

Sadly not that simple. The workaround is needed for all devices but it doesn't work flawlessly on all of them.

Reply:
done